src/client/components/Registration.jsx

```diff
@@ -11,14 +51,44 @@ const Registration = () => {
                         <li>Pick your dream team</li>
                     </ul>
                 </div>
-                <form id="registrationForm">
-                    <label htmlFor="email" className="emailLabel">Email</label>
-                    <input type="text" name="email" id="email" className="regInput"></input>
-                    <label htmlFor="username" className="usernameLabel">Username</label>
-                    <input type="text" name="username" id="username" className="regInput"></input>
-                    <label htmlFor="password" className="passwordLabel">Password</label>
-                    <input type="text" name="password" id="password" className="regInput"></input>
-                    <input type="submit" className="submit" id="submit" value="Create Account"></input>
+                <form id="registrationForm" onSubmit={handleSubmit}>
+                    <label htmlFor="email" className="emailLabel">
+                        Email
+                    </label>
+                    <input 
+                        type="text" 
+                        name="email" 
+                        id="email" 
+                        className="regInput" 
+                        value={userDetails.email}
+                        onChange={handleChange}
+                        required
+                    />
+                    <label htmlFor="username" className="usernameLabel">
+                        Username
+                    </label>
+                    <input 
+                        type="text" 
+                        name="username" 
+                        id="username" 
+                        className="regInput"
+                        value={userDetails.username}
+                        onChange={handleChange}
+                        required
+                    />
+                    <label htmlFor="password" className="passwordLabel">
+                        Password
+                    </label>
+                    <input 
+                        type="text" 
+                        name="password" 
+                        id="password" 
+                        className="regInput"
+                        value={userDetails.password}
+                        onChange={handleChange}
+                        required
+                    />
+                    <input type="submit" className="submit" id="submit" value="Create Account"/>
```

Comment:
This does make it easier to look at. Nice job :). Also controlled forms are always good